spec/spec_helper.rb

```diff
@@ -94,78 +50,29 @@ def redis_service_file(service_name: redis_service_name, service_provider: nil)
   end
 end
 
-def debian_facts
-  {
-    operatingsystem: 'Debian',
-    osfamily: 'Debian',
-    operatingsystemmajrelease: '8',
-    puppetversion: '4.5.2',
-    lsbdistcodename: 'jessie'
-  }
-end
-
-def freebsd_facts
-  {
-    operatingsystem: 'FreeBSD',
-    osfamily: 'FreeBSD',
-    puppetversion: '4.5.2'
-  }
-end
-
-def centos_6_facts
-  {
-    operatingsystem: 'CentOS',
-    osfamily: 'RedHat',
-    operatingsystemmajrelease: '6',
-    puppetversion: '4.5.2'
-  }
-end
-
-def centos_7_facts
-  {
-    operatingsystem: 'CentOS',
-    osfamily: 'RedHat',
-    operatingsystemmajrelease: '7',
-    puppetversion: '4.5.2'
-  }
-end
-
-def debian_wheezy_facts
-  {
-    operatingsystem: 'Debian',
-    osfamily: 'Debian',
-    operatingsystemmajrelease: '8',
-    puppetversion: '4.5.2',
-    lsbdistcodename: 'wheezy'
-  }
-end
-
-def ubuntu_1404_facts
-  {
-    operatingsystem: 'Ubuntu',
-    osfamily: 'Debian',
-    operatingsystemmajrelease: '14.04',
-    puppetversion: '4.5.2',
-    lsbdistcodename: 'trusty'
-  }
-end
-
-def ubuntu_1604_facts
-  {
-    operatingsystem: 'Ubuntu',
-    osfamily: 'Debian',
-    operatingsystemmajrelease: '16.04',
-    puppetversion: '4.5.2',
-    lsbdistcodename: 'xenial'
-  }
-end
-
-def archlinux_facts
-  {
-    operatingsystem: 'Archlinux',
-    osfamily: 'Archlinux',
-    puppetversion: '4.5.2'
-  }
+add_custom_fact :service_provider, ->(os,facts) do
```

Comment:
we should really add this to our modulesync_config or puppetlabs_spec_helper. That's such a commonly used fact.

Reply:
Possibly. I've been thinking about sets of custom facts per module. Maybe `puppetlabs-stdlib` should even provide such a thing.